cl/search/docket_number_cleaner.py

```diff
@@ -30,6 +49,52 @@
     ]
 }
 
+system_prompts = {Court.FEDERAL_APPELLATE: F_PROMPT}
+tie_breaker_prompts = {Court.FEDERAL_APPELLATE: F_TIE_BREAKER}
+
+
+def get_redis_key_prefix() -> str:
+    """Simple tool for getting the prefix for the redis key prefix.
+    Useful for avoiding test collisions.
+    """
+    return "docket_number_cleaning"
+
+
+def get_redis_key() -> str:
+    """Generate a Redis key for storing docket_number_cleaning llm batches.
+    :return: A Redis key string for the llm batch.
+    """
+    return f"{get_redis_key_prefix()}:llm_batch"
+
+
+def update_docket_number(
+    docket_id: int, docket_number: str, start_timestamp: datetime, r: Redis
+) -> None:
+    """
+    Update the docket_number field of a Docket object & remove it from the LLM cleaning Redis set if it hasn't been modified since the start timestamp.
+
+    :param docket_id: The primary key ID of the Docket to update.
+    :param docket_number: The new docket number to set.
+    :param start_timestamp: The timestamp marking the start of the daemon job.
+    """
+    with transaction.atomic():
+        docket = (
+            Docket.objects.select_for_update()
+            .only("id", "docket_number", "date_modified")
+            .filter(id=docket_id)
+            .first()
+        )
+        if not docket:
```

Comment:
In case the docket is missing (for example, if it was removed before the daemon runs), should we remove the `docket_id` from the Redis set so we don’t end up with stale IDs lingering forever?